onadata/libs/serializers/project_serializer.py

```diff
@@ -398,14 +398,14 @@ def update(self, instance, validated_data):
             metadata = dict()
         owner = validated_data.get('organization')
 
-        if self.partial and metadata:
+        if metadata:
             if not isinstance(instance.metadata, dict):
                 instance.metadata = {}
 
             instance.metadata.update(metadata)
             validated_data['metadata'] = instance.metadata
 
-        if self.partial and owner:
```

Comment:
`self.partial` is never set; while calling the serializer from the viewset you need to pass `partial=True` for this to be set. _This code wasn't running previously... I believe this should run whenever the owner is changed regardless if the request is partial or not_